packages/breadboard-ui/src/elements/input/input.ts

```diff
@@ -49,7 +54,12 @@ const parseValue = (type: Schema["type"], input: HTMLInputElement) => {
 };
 
 @customElement("bb-input")
-export class Input extends LitElement {
+export class Input extends LitElement implements BreadboardWebElement {
+  @property()
+  onError = (error: BreadboardElementError) => {
```

Comment:
Is there a reason for having this property? Changing it would cause a re-render, and it seems mostly here for logging. Ideally we just emit an Event and have the containing element handle it, which could include logging it if that's helpful.

Reply:
I thought of a different approach that avoids the interface implementation as well as the onError property on the Lit element. Also, the error event is now emitted as an InputErrorEvent which is a subclass of Event, which should be consistent with the [other event classes](https://github.com/breadboard-ai/breadboard/blob/main/packages/breadboard-ui/src/events/events.ts).

As for the issue of event names being hardcoded (see snippet on https://github.com/breadboard-ai/breadboard/pull/965#discussion_r1512672071), I've addressed this by adding a new "ErrorNames" type, which would be specified in the React wrapper (within the React app) like so:

```typescript
const LitReactInput = createComponent({
		tagName: "bb-input",
		elementClass: Elements.Input,
		react: React,
		events: {
			onerror: ErrorNames.INPUT_ERROR
		},
	});`
	`<LitReactInput onerror={() => console.error("error ocurred")} configuration={{ schema: schema }} />
```

However, there is still a limitation in this approach considering that someone could still hardcode a string as the value for "onerror" and “createComponent” won’t give any indication that doing it is not "allowed". I could not identify any way of forcing the “onError” above to only accept a specific event name.